packages/lit-html/src/parts.ts

```diff
@@ -78,13 +100,17 @@ export const setPartValue = <T extends Part>(
     }
     const newValues = [...(part._value as Array<unknown>)];
     newValues[index] = value;
-    (part as AttributePart)._setValue(newValues, 0);
+    (part as AttributePart)._setValue(newValues, directiveParent, 0);
   } else {
-    part._setValue(value);
+    part._setValue(value, directiveParent);
   }
   return part;
 };
 
+export const updateDirectiveValue = (directive: Directive, value: unknown) => {
```

Comment:
Seems like this might as well be an instance method now. I'd consider moving the whole Directive class here now.

Reply:
Let's handle in a separate PR. Added to https://github.com/Polymer/lit-html/issues/1261.